### PR TITLE
Add pointer event test for cropped element

### PR DESCRIPTION
**Pull request by sherryzy**
b/238885425
b/266714023

### CODE REVIEW

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+        window.onload = () => {
+            setAllHandlers('pointer', '#upper_element', logEvent);
+            setAllHandlers('pointer', '#lower_element', logEvent);
+            setHandlers('click', '#end', endTest);
+            setupFinished();
+        }
+    </script>
+
+    <div class="list">
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:70]: No hard preference, but I like the elements to be above the script and closer to the css

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        <div id="upper_element" class="upper_element"></div>
+        <div id="lower_element" class="lower_element"></div>
+    </div>
+    <div id="end" class="end">End</div>
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:74]: What does this do?

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html [hunk 3]
```diff
@@ -0,0 +1,78 @@
+<!DOCTYPE html>
+
+<html>
+
+<head>
+    <title>Cobalt pointer event test on cropped element</title>
+    <script src='black_box_js_test_utils.js'></script>
+    <script src='pointer_event_test_utils.js'></script>
+    <style>
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:9]: Where is the cropping happening?

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html (hunk 2 above)
Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:74]: It's for ending the test. We perform a click which triggers `endTest()`.

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html (hunk 3 above)
Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:9]: The large bottom part of "upper_element" is being hidden by the "lower_element".

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html (hunk 1 above)
Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:70]: Will address this together with other comments if there are any.

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html (hunk 3 above)
Comment (andrewsavage1) [reply cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:9]: Talked offline, we'll want another test for cropping via `overflow: scroll`, but this is also fine to add

File: cobalt/black_box_tests/testdata/pointer_event_test_utils.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+setTimeout(fail, kTimeout);
+
+function fail() {
+    console.log("Failing due to timeout!");
+    assertTrue(false);
+}
+
+function phaseName(phase) {
+    switch (phase) {
+        case 0: return 'none';
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_test_utils.js:29]: I can't remember if we already talked about this, but I think ideally the case values would also be `Event.NONE`, `Event.CAPTURING`, etc.

File: cobalt/black_box_tests/testdata/pointer_event_test_utils.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        case 2: return 'at target';
+        case 3: return 'bubbling';
+    }
+    return `[unknown: '${phase}']`;
+}
+
+let current_event_number = 0
+let failure_count = 0;
+
+function checkEvent(e) {
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_test_utils.js:40]: This look like an old version of checkEvent unlike the version I suggested?

Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_test_utils.js:40]: This is a stacked PR, and you only need to review the top commit. I'll do rebase shortly.

File: cobalt/black_box_tests/testdata/pointer_event_test_utils.js (hunk 4 above)
Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_test_utils.js:29]: This function phaseName is only for logging, so it's more readable if we just keep it as a string.

Comment (andrewsavage1) [reply cobalt/black_box_tests/testdata/pointer_event_test_utils.js:29]: I mean we could change it to look like `case Event.NONE: return 'none'`

Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_test_utils.js:29]: Oh, it's already changed in my previous PR. This PR is out dated. Rebasing now.

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            // No pointer event on cropper element.
+            { name: 'pointermove', id: 'lower_element', phase: Event.AT_TARGET },
+            // actions.click()
+            { name: 'pointerdown', id: 'lower_element', phase: Event.AT_TARGET },
+            { name: 'pointerup', id: 'lower_element', phase: Event.AT_TARGET },
+        ];
+
+        function endTest(e) {
+            console.log('Ending test.')
+            assertTrue(failure_count == 0);
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:56]: Use `===`

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+            // actions.move_to_element(upper_element).pause(_SLEEP_AFTER_MOVE_TIME)
+            // No pointer event on cropper element.
+            { name: 'pointermove', id: 'lower_element', phase: Event.AT_TARGET },
+            // actions.click()
+            { name: 'pointerdown', id: 'lower_element', phase: Event.AT_TARGET },
+            { name: 'pointerup', id: 'lower_element', phase: Event.AT_TARGET },
+        ];
+
+        function endTest(e) {
+            console.log('Ending test.')
```
Comment (andrewsavage1) [inline cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:55]: Does this log add anything? Is it not already clear?

**Review by andrewsavage1 — APPROVED** [review]

File: cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html (hunk 7 above)
Comment (sherryzy) [reply cobalt/black_box_tests/testdata/pointer_event_on_cropped_element_test.html:55]: Well, it provides some value (like we clicked the end element) but not a lot. I'm copying from the existing test. Will delete it.
